cpmpy/solvers/pysat.py

```diff
@@ -285,9 +296,51 @@ def _post_constraint(self, cpm_expr):
 
                 # post the clauses
                 self.pysat_solver.append_formula(clauses)
-            else:
-                raise NotImplementedError(f"Non-operator constraint {cpm_expr} not supported by CPM_pysat")
 
+            # WEIGHTED !
+            elif isinstance(left, Operator) and (left.name in ["wsum", "mul"]) and is_int(bound):
+
+                if not CPM_pysat.pb_supported():
+                    raise ImportError("Please install PyPBLib: pip install pypblib")
+                from pysat.pb import PBEnc
+
+                if left.name == "mul":
+                    # single weight,value pair, in list
+                    weights = [left.args[0]]
+                    lits = [self.solver_var(left.args[1])]
+                else: # wsum
+                    weights = left.args[0]
+                    lits = self.solver_vars(left.args[1])
+
+                clauses = []
+                if cpm_expr.name == "<" or (cpm_expr.name == "!=" and bound >= len(lits)):
```

Comment:
You are missing the weights in the condition.
Now `10 * a + 5 * b != 5` is converted to `10 * a + 5 * b <= 4`

Reply:
current version generates '[(BV0) -> (sum([10, 5] * [a, b]) <= 4), (~BV0) -> (sum([10, 5] * [a, b]) >= 6)]'
which I think is correct? @IgnaceBleukx can you resolve this thread if you agree?

Reply:
OK, I was confused here... it is the 'linearize' transfo that I added today for testing that did that rewriting.
You mean that it should be bound >= sum(weights) I think... but then indeed what about negatives (your comment lower). We could do bound >= sum(max(0,weights)) but what are we doing here then...

Either the bound is that max, and you can just post it as <= bound-1, or the bound is higher then the max and then the constraint is always true...

Maybe we should just drop this 'optimisation'?
I'm hoping that the PB encoding itself will check the 'always true' edge cases and so cover this appropriately?